grakn-engine/src/main/java/ai/grakn/engine/controller/response/KeyspaceResponse.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.controller.response;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.auto.value.AutoValue;
+
+import javax.annotation.CheckReturnValue;
+
+/**
+ * <p>
+ *     Response object representing {@link ai.grakn.Keyspace}
+ * </p>
+ *
+ * @author Filipe Peliz Pinto Teixeira
+ */
+@AutoValue
+public abstract class KeyspaceResponse {
```

Comment:
It looks like `grakn-core` already has a Jackson dependency (since `TaskId` seems to be using it). Is this class definitely necessary in that case?

Reply:
Well that's the main design decision. Should we embed the annotations or should we create new objects rep represent them. Think of the concepts for example?

I keep swinging back and forth on this one.

Reply:
Right now we're doing a hybrid approach, where `TaskId` are using the original approach and `Keyspace` isn't. I don't know which is right yet, but I don't think we should mix it up unless we have to.

Reply:
> I keep swinging back and forth on this one.

As is evident  by those issues. In `TaskId ` I can "justify" it because it exists within the context of engine where the Jackson dependency is supposed to live.

With the `keyspace` object that is not the case.

So ya not really sure. I am leaning more toward embedding them in the original objects but the affect on concepts may be kind of shitty. . . .